storage/innobase/os/os0enc.cc

```diff
@@ -1856,15 +1856,35 @@ ulint Encryption::get_master_key_id() { return s_master_key_id; }
 bool Encryption::dblwr_encrypt_page(fil_space_t *space, page_t *in_page,
                                     file::Block_ptr &enc_block,
                                     ulint &enc_block_len) {
-  if (!FSP_FLAGS_GET_ENCRYPTION(space->flags)) {
+  if (!FSP_FLAGS_GET_ENCRYPTION(space->flags) &&
```

Comment:
Shouldn't this be || condition?

Reply:
It should be &&. This is because we should not only encrypt pages that have encryption flag set, but also pages that do not have encryption flag set, but do have crypt_data assigned and it is "in encrypting" mode (being encrypted by encryption threads). This condition describes when we should skip encrypting page in double write buffer.

Reply:
Ok, can you explain this in a comment. When would this scenario happen? ie. we have space->crypt_data but do not write encrypted. is this online to unencrypted scenario?

Reply:
@robgolebiowski ^

Reply:
@robgolebiowski hope you can add a comment to code